src/main/scala/za/co/absa/commons/lang/CollectionImplicits.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.commons.lang
+
+object CollectionImplicits {
+
+  implicit class IteratorOps[A](iter: Iterator[A]) {
```

Comment:
Could this be made into value class? That would avoid creation of unnecessary instances durring runtime.

https://docs.scala-lang.org/overviews/core/value-classes.html#extension-methods

Reply:
done